sklearn/utils/validation.py

```diff
@@ -324,21 +324,26 @@ def check_array(array, accept_sparse=None, dtype="numeric", order=None, copy=Fal
     if isinstance(accept_sparse, str):
         accept_sparse = [accept_sparse]
 
+    numeric = dtype == "numeric"
+
     if sp.issparse(array):
-        if dtype == "numeric":
+        if numeric:
```

Comment:
Will do. The point is to "safe" the original value of dtype after possibly setting it to "None" later. Maybe the logic is not that great. Will add a comment, too.